README.md

```diff
@@ -22,22 +23,19 @@
 * Resolution should not depend on file extensions, allowing ESM syntax in `.js` files.
 * The directory structure of a module should be treated as private implementation detail.
 
-## `package.json` Interface
+## `package.json` Interfaces
 
-We propose a field in `package.json` to specify one or more entrypoint locations when importing bare specifiers.
+We propose two fields in `package.json` to specify entrypoints and internal aliasing of bare specifiers - [`"exports"`](#1-exports-field) and [`"imports"`](#2-imports-field).
```

Comment:
I would very much prefer these to be separated; ie, i want to be sure exports can land unflagged in both ESM and cjs with or without imports. (if including imports here won’t interfere with that, consider this comment resolved)

Reply:
I consider making a decision on these semantics a blocker for unflagging `exports`, no matter if it's in the same repo or not. Because likely `imports` will have higher priority and we should know how that could affect upgrade paths before unflagging `exports` without `imports`.

Reply:
Can you explain “higher priority”? i thought that external users are the only ones to see exports, and internal users are the only ones to see imports.

Reply:
I assume the following *may* work, depending on where we end up with mandatory prefixes:

```js
// client/a.mjs
import 'x/zapp';

// client/package.json
({
  "imports": { "x/zapp": "x/zup" }
})
```

```js
// x/package.json
({
  "exports": {
    "./zapp": "./lib/zapp.mjs",
    "./zup": "./lib/zup.mjs"
  }
})
```

Imports applies first, before `exports` is applied. So I would assume that the import in `a.mjs` resolves to `file:///path/to/x/lib/zup.mjs`. In that way imports has a higher priority.

Reply:
Ah, I agree they'd apply first. However I'm not sure why there'd be any semantic conflict; by the time the attempt hits `exports`, `imports` has already been processed, applied, and discarded.

Reply:
There have been designs for `imports` that would've overwritten `exports` as opposed to forwarding to it. So yes - with the current design @guybedford worked out, we may be able to ship `imports` after `exports`. Above I only said that the semantics need to be somewhat figured out to make sure that we *can* unflag `exports` without making the addition of `imports` a breaking change. *Shipping* `imports` isn't the blocker I raised.

Reply:
Would it help to add a note that these fields _can_ be implemented / supported independently?

Reply:
Yes, it would :-)

Reply:
Ok, added a note to this effect.